### ISSUE TITLE
Minecraft freezes when I turn computer on

### ISSUE THREAD

**Issue by pgolinski** [MightyPirates/OpenComputers#369]
I play new modded world from some time without any problem. Recently I decided to try OpenComputer features. When I click turn on button, Minecraft freezes immediately or after several seconds. Sometimes it says "terminating internal server". I have no crash reports or logs. I noticed that game when freezes still takes CPU at level of 80%.

I'll very thankful for any clue how to track down problem (or repair it if it is only OpenComputers fault ;)). I realy care to have this mod working.

Minecraft: 1.6.4
Forge: 9.11.1.953
OpenComputers: 1.2.13.379
Mods:
BiomesOPlenty-1.2.1.416
buildcraft-4.2.2
ChickenChunks 1.3.3.4
Factorization-0.8.34
forestry-2.3.1.1
ic2: 2.0.397
InventoryTweaks-1.56-b77
ironchest-5.4.1.677
Natura-2.1.14
Nei-1.6.1.9
ProjectRed-4.3.5.30
Railcraft-8.4.0.0
ThermalExpansion-3.0.0.7
TreeCapitator
Waila
Rei's minimap

**Comment by fnuecke** [comment]
Hmm, does this also happen when no other mods are present?
Are there any blocks from other mods _next_ to the computer you're turning on?

**Comment by pgolinski** [comment]
Thanks for your reply. I tested on two brand new creative worlds. At first I used Creative Computer Case with monitor and keyboard. Second I used Tier 1 Computer case with monitor and keyboard powered by IC2 cables. Same effect. I will make clean environment with only OpenComputers installed and I let you know about results.

**Comment by pgolinski** [comment]
I run OpenComputers with clean environment and its working so far. I will install mods one-by-one to find that causing problems. I will let you know.

**Comment by fnuecke** [comment]
Thank you, I appreciate it!

**Comment by black3agl3** [comment]
Not sure if this is related to your problem but I had some crazy freezes when I tried to turn on a computer. I would have to basically kill minecraft using task manager. There were lots of server can't keep up lines in the log.

Setting -XX:MaxPermSize=256m solved my problem though.

**Comment by pgolinski** [comment]
Well, I reinstalled all mods and everything is working almost perfectly (Minecraft freezed two times, but in different circumstances). Thats weird. I used same files - just moved them from dir to dir and didn't reinstall forge. I found strange thing in forge logs: "This world was saved with mod ForgeMultipart version 1.0.0.250 and it is now at version 1.0.0.244, things may not work well". This was when I loaded my survival world. It looks like forge downgrade its module itself. I don't know. 
Nevertheless I added JAVA_OPT environment variable with setting just like black3agl3 suggested (thanks!), but not played Mincraft since then. I would leave open this issue for a while if you don't mind.

**Comment by fnuecke** [comment]
Sure. If someone _can_ reproduce it, please let me know. Thanks.

**Comment by fnuecke** [comment]
Been quite a while now without new sightings, closing.

**Comment by pgolinski** [comment]
I was just about to post. I played some without any problem, and day before yesterday I installed OpenComponents and updated OpenComputers. My Minecraft freezes again :(. I'm still trying to diagnoze it.

**Comment by fnuecke** [comment]
Hrm, assuming it's a dead-lock and not an out of memory issue, try having a look at the state of the threads with VisualVM (if you have the JDK it should be installed with that).

**Comment by asiekierka** [comment]
What operating system and Java version? Did you apply any significant modifications to your OS?

**Comment by pgolinski** [comment]
@fnuecke No it isn't a lack of memory. I have Intel I5 with 8GB of RAM. Hard drive is SSD
@asiekierka Windows 7 64bit. Currently I have Java 7 update 65. 

That is thing that I just CAN'T UNDERSTAND. I fought with it yesterday for an hour. Updated Java and my video drivers (GeForce 9800GT), give a try for different java start arguments. Nothing changed.. Today I installed Java JDK and VisualVM as suggested. I start Minecraft with VisualVM on screen and what? Everything goes even smoother than before! 80-120FPS. Could installation of JDK make any difference? 
I installed openOS and wrote 'hello world' scirpt. Everything fine. My time is ran out for now, but I hope that after few days if I want play minecraft it will be still working.

**Comment by fnuecke** [comment]
I'm not suggesting your _system_ doesn't have enough RAM, I'm talking about how much Java may allocate, for the heap (-Xmx) but also in particular for permgen (-XX:MaxPermSize). It's intriguing though that you'd actually get _better_ performance with the JDK's java binary. I can only guess that maybe it wasn't using the Java installation you thought it was using, but an older one instead?

**Comment by pgolinski** [comment]
I set these values to 2048M. I think it's should be enough.
You're propably right about java version. I found later an entry in windows events, which says that it couldn't delete old java version :/. Maybe it was succeded after computer restarts.
I was really frustrated because of lack of information what is going on. You see I don't have much time and must rely more on intuition rather than scrupulous research.
Anyway I thank you for your help and I'm happy to see that man like you developing my favorite mod. 
This issue may be closed now and forever I hope ;)
